src/kpro.erl

```diff
@@ -357,50 +359,61 @@ encode({{array, T}, L}) when is_list(L) ->
 encode({array, L}) when is_list(L) ->
   Length = length(L),
   [<<Length:32/?INT>>, [encode(I) || I <- L]];
-encode(#kpro_PartitionMessageSet{} = R) ->
+%% FIXME encoder could be better? also what about  ...v1_data and ...v2_data
+%% probably should be done with encode(records, Bin) for RECORDS type)
+encode(#kpro_produce_request_v0_data{} = R) ->
   %% messages in messageset is a stream, not an array
-  EncodedMessages = R#kpro_PartitionMessageSet.message_L,
+  %% FIXME what is the difference between stream and array here?
```

Comment:
Arrays in kafka protocol are preceded by int32 containing number of elements.
In fetch response message sets are preceded by int32 containing byte size of the set.